### DISCUSSION TITLE
Using DMA with more than one slave?

### DISCUSSION THREAD

**Discussion by Zombieschannel** [Bodmer/TFT_eSPI#2738] [Q&A - General]
Wanted to ask if it is possible to use DMA functionality from this library if there is more than one slave in SPI.

For example:
I have an ST7735 display with an SD card module connected to an ESP32. 
I am aware that ESP32 has both HSPI and VSPI, but because of some other reasons, I can only use VSPI. I got this setup to work without using DMA. If DMA is enabled, the display does work but the sd card slot does not. 
First I init the display, do a test clear screen, and then set up the SD card. Before setting up the SD card, I also called both .dmaWait(); and .endWrite(); to make sure that the display would not be in use. 

I even tried to deInitDMA() before setting up the sd card thinking maybe it is possible to disable DMA every time I would need to access the sd card. No luck there. 
So is there any way to achieve this?
Any help is appreciated.

**Comment by Bodmer** [comment]
That should work. You would need to manage the chip select within the sketch and ensure the transactions are complete before changing the selected display.
